modules/imgproc/src/sumpixels.simd.hpp

```diff
@@ -138,7 +138,7 @@ struct Integral_SIMD<uchar, int, double>
                     v_expand(el8, el4l, el4h);
                     el4l += prev;
                     el4h += el4l;
-                    prev = v_broadcast_element<v_int32::nlanes - 1>(el4h);
+                    prev = v_broadcast_element<v_int32::max_nlanes - 1>(el4h);
```

Comment:
I think, it was meant originally that the highest-significant element should be populated. The new code will likely give incorrect results when `max_nlanes != nlanes`. Some different solution needs to be implemented here

Reply:
This can probably be solved by defining a macro in intrin_x.hpp just like how `v_broadcast_element` is defined. We then make use of `v_broadcast_element<_Tpvec::nlanes - 1>(v)` in our newly defined `v_broadcast_last_element(v)` function and use it in place of `v_broadcast_element`. See below


```
#define OPENCV_HAL_IMPL_NEON_BROADCAST_LAST(_Tpvec) \
inline _Tpvec v_broadcast_last_element(_Tpvec v) \
{ \
    return v_broadcast_element<_Tpvec::nlanes - 1>(v); \
}
```